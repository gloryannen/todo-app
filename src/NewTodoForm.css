.newTodoFormInput {
  padding: 4px;
  border-radius: 0px;
  border: none;
  color: black;
}

.newTodoFormAddBtn {
  padding: 4px;
  font-weight: bold;
  border: none;
  background-color: white;
  cursor: pointer;
}

h1 {
  color: white;
  text-align: center;
  font-size: 3rem;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande',
    'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}
