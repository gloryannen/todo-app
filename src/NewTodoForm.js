import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import './NewTodoForm.css';

const NewTodoForm = ({ addTodo }) => {
  const [todo, setTodo] = useState('');

  // Handle input values
  const handleChange = (e) => {
    setTodo(e.target.value);
  }

  // Handle form submission - create new todo , assign id, clear input
  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ todo, id: uuid() });
    setTodo(' ');
  };

  //Todo form
  return (
    <div className='newTodoFormContainer'>
      <h1>Todo App</h1>
      <form onSubmit={ handleSubmit }>
        <input className='newTodoFormInput'
          type='text'
          id='todo'
          name='todo'
          value={ todo }
          onChange={ handleChange }
          placeholder='Add a task'
          required
        />
        <button className='newTodoFormAddBtn'>+</button>
      </form>
    </div>
  )
}

export default NewTodoForm;