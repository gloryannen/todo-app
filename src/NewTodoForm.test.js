import React from 'react';
import '@testing-library/jest-dom';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

// Smoke test
it("renders without crashing", function () {
  render(<NewTodoForm />);
});

// Snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});