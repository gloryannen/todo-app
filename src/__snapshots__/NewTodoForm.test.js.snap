// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="newTodoFormContainer"
  >
    <h1>
      Todo App
    </h1>
    <form>
      <input
        class="newTodoFormInput"
        id="todo"
        name="todo"
        placeholder="Add a task"
        required=""
        type="text"
        value=""
      />
      <button
        class="newTodoFormAddBtn"
      >
        +
      </button>
    </form>
  </div>
</DocumentFragment>
`;
